gen/src/main.rs

```diff
@@ -231,21 +250,51 @@ fn main() {
     eprintln!("All bindings generated!");
 }
 
-fn git_checkout(rev: &str) {
-    // Delete any generated files from previous versions.
+fn git_init() {
+    // Clone the linux kernel source repo if necessary. Ignore exit code as it will be non-zero in
+    // case it was already cloned.
+    // Use a treeless partial clone to save disk space and clone time.
+    // See https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/ for
+    // more info on partial clones.
+    // Note: this is not using the official repo
+    // git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git but the github fork as the
+    // server of the official repo doesn't recognize filtering.
+    Command::new("git")
+        .arg("clone")
+        .arg("https://github.com/torvalds/linux.git")
+        .arg("--filter=tree:0")
+        .arg("--no-checkout")
+        .status()
+        .unwrap();
```

Comment:
Instead of ignoring the exit code, this should test for the existence of gen/linux/.git or so up front, since git clone can fail for other reasons and we don't want the CI lumbering on and reporting confusing errors.